zerver/lib/actions.py

```diff
@@ -1517,16 +1518,6 @@ def do_send_messages(messages_maybe_none: Sequence[Optional[MutableMapping[str,
     # Filter out messages which didn't pass internal_prep_message properly
     messages = [message for message in messages_maybe_none if message is not None]
 
-    # Filter out zephyr mirror anomalies where the message was already sent
-    already_sent_ids: List[int] = []
-    new_messages: List[MutableMapping[str, Any]] = []
-    for message in messages:
-        if isinstance(message['message'], int):
-            already_sent_ids.append(message['message'])
-        else:
-            new_messages.append(message)
-    messages = new_messages
```

Comment:
This code was added by commit 9c8a9ac9479a55029c5776f3a3cfb0d08c017490 with message “This was previously causing us to generate a traceback every time we hit a duplicated zephyr due to CC'ing.” Do we expect these duplicated Zephyrs to no longer occur?

Reply:
@andersk I have updated the PR. I have changed the caller of check_message to return the message-id directly when the exception is raised instead of returning it via do_send_messages as we were doing it previously to avoid using `MessageInfo` dataclass for this case.
The checks are failing because of coverage issue and I will fix this. Does this approach seem fine to you or we should do something else?

Reply:
Also just wanted to confirm, we would not these checks for `check_send_stream_message` and `check_send_private_message` as these are just used by webhooks and tests, and from what I have read in the Integrations section of https://zulip.readthedocs.io/en/latest/subsystems/client.html, I can understand that client cannot be `zephyr_mirror` for messages sent through webhooks. So, the `try..except` block will be needed in `check_send_message` and `check_schedule_message` only and can be removed from other places.